### DISCUSSION TITLE
SSL_CTX_set1_param not found

### DISCUSSION THREAD

**Discussion by yukimasa-nomoto** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1741] [Q&A] [answered]
HI There 

i use the this sdk after Cross-Compliling

ubuntu 18 is compiling machine
target is arm7  built in yocto

i started below
-----------------------------------
cmake .. -DBUILD_TEST=TRUE -DBUILD_STATIC_LIBS=TRUE -DBUILD_OPENSSL=TRUE -DBUILD_OPENSSL_PLATFORM=linux-armv4 -DOPENSSL_EXTRA="--cross-compile-prefix=\"\";no-async;no-asm" 
-----------------------------------

openssl is built cleary
but websockets can not build cleary

-------------------------------
-- Performing Test LWS_HAVE_TCP_USER_TIMEOUT
-- Performing Test LWS_HAVE_TCP_USER_TIMEOUT - Success
Compiling with SSL support
-- Found OpenSSL: /home/work/webrtc/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/lib/libcrypto.a (found version "1.1.1g") 
OpenSSL include dir: /home/work/webrtc/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/include
OpenSSL libraries: /home/work/webrtc/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/lib/libssl.a;/home/ywork/webrtc/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/lib/libcrypto.a
-- Looking for SSL_CTX_set1_param
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LIBCAP_LIBRARIES
    linked by target "cmTC_8bcf9" in directory /home/work/webrtc/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/libwebsockets/build/src/project_libwebsockets-build/CMakeFiles/CMakeTmp
------------------------------  

argument is less ? mistaken?
my openssl by cross compiled is wrong?

anyone can help?

**Comment by niyatim23** [comment, marked as answer]
We have three cross compilation builds in our CI [here](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/b820e2ea3d28c670da2fbc789face742cb949ef1/.github/workflows/ci.yml#L327-380) that work without any issues. Please follow the steps here and see if it works for you

**Comment by yukimasa-nomoto** [comment]
[niyatim23]
Thank you for your Great Answer

I createde new Environment on another Vmware
I tried to build by using [ci.yml]
That worked Fine

you're right

my Build Environment on ubuntu16.04 for Yocto must be worse

I retry to create enviroment for yocto.
